pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -687,5 +687,50 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
             arr[k] *= inv_norm;
         }
     }
+
+    /**
+     * @brief Prepares a single computational basis state.
+     *
+     * @param index Index of the target element.
+     */
+    void setBasisState(const std::size_t index) {
+        auto length = this->getLength();
+        PL_ABORT_IF(index > length - 1, "Invalid index");
+
+        auto *arr = this->getData();
+        std::fill(arr, arr + length, 0.0);
+        arr[index] = {1.0, 0.0};
+    }
+
+    /**
+     * @brief Set values for a batch of elements of the state-vector.
+     *
+     * @param values Values to be set for the target elements.
+     * @param indices Indices of the target elements.
+     */
+    void setStateVector(const std::vector<std::size_t> &indices,
+                        const std::vector<ComplexT> &values) {
+        auto num_indices = indices.size();
+        PL_ABORT_IF(num_indices != values.size(),
+                    "Indices and values length must match");
+
+        auto *arr = this->getData();
+        auto length = this->getLength();
+        std::fill(arr, arr + length, 0.0);
+        for (std::size_t i = 0; i < num_indices; i++) {
+            PL_ABORT_IF(i >= length, "Invalid index");
+            arr[indices[i]] = values[i];
+        }
+    }
+
+    /**
+     * @brief Reset the data back to the \f$\ket{0}\f$ state.
+     *
+     */
+    void resetStateVector() {
```

Comment:
Is this a noexcept method?
```suggestion
    void resetStateVector() noexcept {
```

Reply:
It is calling `setBasisState` which can throw.

Reply:
Same.